scripts/version.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+if [ $# -eq 1 ]; then
+	VERSIONSTR="${1}"
+else
+    echo "usage: $0 <version>, where version needs to be a 4 part version number, eg. 2023.1.1.1046"
+    exit 1
+fi
+
+VERSIONARR=(${VERSIONSTR//./ })
+
+echo "setting version to"
+echo "major: ${VERSIONARR[0]}"
+echo "minor: ${VERSIONARR[1]}"
+echo "debug: ${VERSIONARR[2]}"
+echo "patch: ${VERSIONARR[3]}"
+
+perl -pi -e "s,extern const int MAJOR_VERSION = .*;,extern const int MAJOR_VERSION = ${VERSIONARR[0]};,g" src/libs/vmisc/projectversion.cpp
```

Comment:
Please add comments to describe how & when the Major and Minor versions are updated.

Reply:
Will add more documentation. The idea here in short: instead of manually editing the version in multiple places, just run the script with the desired version number, eg 2023.1.3.2129:

./scripts/version.sh 2023.1.3.2129

So the script does not increment the version based on rules, it just takes the new full version number and modifies it in all places. I will add this to the script to make it more clear how to use it. 

A long term possibility would be to have the weekly release job also modify the version with this script so that the version that is visible on the github page is also visible in the application in the about dialog.